src/renderer/io.ts

```diff
@@ -280,16 +229,17 @@ export const preprocessPdfs = (
 
     let pagesOfText = await loadPageJson(dir, "textToDisplay");
 
-    // TODO (Xin): after parsing per-page textToDisplay from PDF pages,
+    // after parsing per-page textToDisplay from PDF pages,
     // auto-grab info from the pdf, including two steps:
     // 1) API communication to python
     // and 2) write to "metadataToHighlight.json"
-    // await createAutoGrabInfo(
-    //   pagesOfText,
-    //   path.join(dir, "metadataToHighlight.json"),
-    //   pdf,
-    //   true //TODO (Xin) later, change to variable overwrite
-    // );
+    await createAutoGrabInfo(
```

Comment:
I like that this is one func call, but it needs a feature flag check and some io safey such as try/catch for checking that the file exists.

Reply:
Sure, I will add that shortly in the afternoon.

Reply:
io safety check inside 
https://github.com/xeniaqian94/lab/blob/b90dd850911d26bb003870c52b35b359c904cbd0/src/renderer/AutoGrab.tsx#L50-L60